src/blockchain/shelley/block.js

```diff
@@ -166,7 +167,21 @@ export default class ShelleyBlock implements Block {
       fragment.free()
     }
     fragments.free()
+
+    const leaderIdPtr = block.leader_id()
```

Comment:
I do not use `shelleyUtils.consumeIdToHex(block.leader_id())` because `leader_id()` return `undefined` for some blocks. Also `LeaderId` object do not have `as_bytes()` method, only `to_string()`